lib/ansible/plugins/become/su.py

```diff
@@ -139,15 +139,22 @@ class BecomeModule(BecomeBase):
         '口令',
     ]
 
-    def check_password_prompt(self, b_output):
+    def check_password_prompt(self, b_output: bytes) -> bool:
         """ checks if the expected password prompt exists in b_output """
-
         prompts = self.get_option('prompt_l10n') or self.SU_PROMPT_LOCALIZATIONS
-        b_password_string = b"|".join((br'(\w+\'s )?' + to_bytes(p)) for p in prompts)
+        password_prompt_strings = "|".join(re.escape(p) for p in prompts)
         # Colon or unicode fullwidth colon
-        b_password_string = b_password_string + to_bytes(u' ?(:|：) ?')
-        b_su_prompt_localizations_re = re.compile(b_password_string, flags=re.IGNORECASE)
-        return bool(b_su_prompt_localizations_re.match(b_output))
+        prompt_pattern = to_bytes(rf"(?:{password_prompt_strings})\s*[:：]")
+        match = re.match(prompt_pattern, b_output, flags=re.IGNORECASE)
+
+        if match:
+            self.prompt = to_text(match.group(0))  # preserve the actual matched string so we can scrub the output
+
+        return bool(match)
+
+    def expect_prompt(self) -> bool:
+        """A password is always required; base impl hangs until timeout if a password is not provided."""
+        return True
```

Comment:
A password is not always required with `su`, but that requires `pam` configuration and is probably very rare